consensus/istanbul/core/core.go

```diff
@@ -304,6 +308,12 @@ func (c *core) startNewRound(round *big.Int) {
 	var newView *istanbul.View
 	var roundChangeCertificate istanbul.RoundChangeCertificate
 	var request *istanbul.Request
+
+	var valSet istanbul.ValidatorSet
+	if c.current != nil {
```

Comment:
When is `c.current` == nil?

Reply:
The first time we call `startNewRound()` which happens inside `Start()`. Actually this happens before even we start listening to events. 

After that `c.current` is never nil (until we call `Stop()`